### PR TITLE
Sort includes in criu/img-*.c

### PR DESCRIPTION
**Pull request by rst0git**
Sort and remove unused/unnecessary include stements in `criu/img-*.c`

### CODE REVIEW

File: criu/img-remote.c [hunk 1]
```diff
@@ -1,26 +1,12 @@
-#include <unistd.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <sys/types.h>
-#include <sys/socket.h>
```
Comment (avagin) [inline criu/img-remote.c]: In this file, sys/socket.h is included here because socket() is called in this file.

Comment (avagin) [reply criu/img-remote.c]: We should not rely on criu headers in such cases. Maybe we should not include linux headers in criu headers...

Comment (rst0git) [reply criu/img-remote.c]: I agree that headers such as `sys/socket.h` and `sys/epoll.h` are used in this file and it is good to have them listed at the beginning as a hint about what functionalities are being used in the implementation that follows. However, if we were to do that consistently we will end up with a lot of duplicated code between *.c files and it is likely (in the future) for a developer to forget to remove an include statement when they remove a specific function call.
@avagin Considering both pros and cons, which headers do you think should be kept?
(I am happy to update the patch.)

Comment (avagin) [reply criu/img-remote.c]: > However, if we were to do that consistently we will end up with a lot of duplicated code between *.c files and it is likely (in the future) for a developer to forget to remove an include statement when they remove a specific function call.

Do you know where sys/socket.h is included now? Is it a side effect? Does the same side effect exists in other libc-s?

File: criu/img-remote.c [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-#include "xmalloc.h"
-#include "criu-log.h"
+#include <sys/un.h>
+#include <unistd.h>
+
 #include "img-remote.h"
-#include "images/remote-image.pb-c.h"
-#include "protobuf-desc.h"
-#include <fcntl.h>
-#include "servicefd.h"
```
Comment (avagin) [inline criu/img-remote.c]: get_service_fd is used in this file

File: criu/img-remote.c [hunk 3]
```diff
@@ -1,26 +1,14 @@
-#include <unistd.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <sys/types.h>
+#include <netinet/in.h>
 #include <sys/socket.h>
 #include <sys/epoll.h>
-#include <netinet/in.h>
-#include <netdb.h>
-#include "xmalloc.h"
```
Comment (avagin) [inline criu/img-remote.c]: xmalloc is used in this file

File: criu/img-proxy.c [hunk 4]
```diff
@@ -1,10 +1,7 @@
 #include <unistd.h>
 
-#include "img-remote.h"
-#include "criu-log.h"
```
Comment (avagin) [inline criu/img-proxy.c]: pr_info, pr_perror are used in this file

**Pull request merged**